bot/exts/info/subscribe.py

```diff
@@ -0,0 +1,149 @@
+ASSIGNABLE_ROLES = (
+    (constants.Roles.announcements, None),
+    (constants.Roles.pyweek_announcements, None),
+    (constants.Roles.advent_of_code, 12),
+    (constants.Roles.lovefest, 2),
+)
+ITEMS_PER_ROW = 3
+
+log = get_logger(__name__)
+
+
+class RoleButtonView(discord.ui.View):
+    """A list of SingleRoleButtons to show to the member."""
+
+    def __init__(self, member: discord.Member):
+        super().__init__()
+        self.interaction_owner = member
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Ensure that the user clicking the button is the member who invoked the command."""
+        if interaction.user != self.interaction_owner:
+            await interaction.response.send_message(
+                ":x: This is not your command to react to!",
+                ephemeral=True
+            )
+            return False
+        return True
+
+
+class SingleRoleButton(discord.ui.Button):
+    """A button that adds or removes a role from the member depending on it's current state."""
+
+    ADD_STYLE = discord.ButtonStyle.success
+    REMOVE_STYLE = discord.ButtonStyle.secondary
+    LABEL_FORMAT = "{action} role {role_name}"
+    CUSTOM_ID_FORMAT = "subscribe-{role_id}"
+
+    def __init__(self, role: discord.Role, assigned: bool, row: int):
+        super().__init__(
+            style=self.REMOVE_STYLE if assigned else self.ADD_STYLE,
+            label=self.LABEL_FORMAT.format(action="Remove" if assigned else "Add", role_name=role.name),
+            custom_id=self.CUSTOM_ID_FORMAT.format(role_id=role.id),
+            row=row,
+        )
+        self.role = role
+        self.assigned = assigned
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Update the member's role and change button text to reflect current text."""
+        if isinstance(interaction.user, discord.User):
+            await interaction.message.delete()
+            self.view.stop()
+            return
+
+        await members.handle_role_change(
+            interaction.user,
+            interaction.user.remove_roles if self.assigned else interaction.user.add_roles,
+            self.role,
+        )
+
+        self.assigned = not self.assigned
+        await self.update_view(interaction)
+        await interaction.response.send_message(
+            self.LABEL_FORMAT.format(action="Added" if self.assigned else "Removed", role_name=self.role.name),
+            ephemeral=True,
+        )
+
+    async def update_view(self, interaction: Interaction) -> None:
+        """Updates the original interaction message with a new view object with the updated buttons."""
+        self.style = self.REMOVE_STYLE if self.assigned else self.ADD_STYLE
+        self.label = self.LABEL_FORMAT.format(action="Remove" if self.assigned else "Add", role_name=self.role.name)
+        view_list = [
+            self if button.custom_id == self.custom_id else button
```

Comment:
What difference does this ternary expression make?

When `button.custom_id == self.custom_id` is True then `button is self` should be True?

Reply:
https://github.com/python-discord/bot/pull/1868/commits/b5a316d7371713801e429490eb5152b0a0e4f646